BraveRewardsUI/Wallet/Notifications/NotificationHandler.swift

```diff
@@ -0,0 +1,192 @@
+  }
+  
+  @objc private func notificationAdded(_ notification: Notification) {
+    // TODO: Filter notification?
+    if currrentNotification == nil {
+      loadNotification()
+    }
+  }
+  
+  private func loadNotification() {
+    guard delegate != nil else {
+      return
+    }
+    if let currrentNotification = currrentNotification {
+      ledger.clearNotification(currrentNotification)
+      self.currrentNotification = nil
+    }
+    
+    //.failedContribution, .invalid & .backupWallet are not supported right now
+    // TODO: Verify that notifications are ordered & they are displayed in FIFO.
+    let ignoredRewardNotificationTypes: [RewardsNotification.Kind] = [.failedContribution,
+                                                                      .invalid,
+                                                                      .backupWallet]
+    let notification = ledger.notifications
+    if let notification = notification.first(where: {
+      !ignoredRewardNotificationTypes.contains($0.kind)
+    }) {
+      currrentNotification = notification
+      switch notification.kind {
+      case .autoContribute, .tipsProcessed, .verifiedPublisher:
+        addMessageNotification(notification: notification)
+      case .grant, .grantAds, .insufficientFunds, .pendingNotEnoughFunds, .adsLaunch:
+        addActionNotification(notification: notification)
+      default:
+        return
+      }
+    } else {
+      delegate?.hideNotifications()
+    }
+  }
+  
+  private func addActionNotification(notification: RewardsNotification) {
+    let category: WalletActionNotification.Category
+    let body: String
+    
+    switch notification.kind {
+    case .grant:
+      body = Strings.NotificationGrantNotification
+      category = .grant
+    case .grantAds:
+      body = Strings.NotificationEarningsClaimDefault
+      category = .grant
+    case .insufficientFunds:
+      body = Strings.NotificationInsufficientFunds
+      category = .insufficientFunds
+    case .pendingNotEnoughFunds:
+      body = Strings.NotificationPendingNotEnoughFunds
+      category = .insufficientFunds
+    case .adsLaunch:
+      body = Strings.NotificationBraveAdsLaunchMsg
+      category = .adsLaunch
+    default:
+      loadNotification()
+      return
+    }
+    let date = Date(timeIntervalSince1970: notification.dateAdded)
+    let notificationView = WalletActionNotificationView(
+      notification: WalletActionNotification(
+        category: category,
+        body: body,
+        date: date))
+    notificationView.closeButton.addTarget(self, action: #selector(tappedNotificationClose), for: .touchUpInside)
+    notificationView.actionButton.addTarget(self, action: #selector(tappedNotificationAction), for: .touchUpInside)
+    delegate?.show(notificationView: notificationView)
+  }
+  
+  @objc private func tappedNotificationClose() {
+    loadNotification()
```

Comment:
Honestly the fact that we're just calling `loadNotification` to clear the current one is kinda weird to me. Can we just split these out or something?

Reply:
Clearing a notification is always followed by loading next notification. This is coupled so much that clearing a notification in itself does not have a useful case. Also the fact if clear and then set a notification, the animation is not linear but abrupt. (cases inside `setNotificationView`)

Refactoring `loadNotification` to `loadNextNotification` and creating a separate method for clearing notification and replacing it for the clear notification code in `loadNotification`